scripts/zones/Lower_Jeuno/npcs/Vingijard.lua

```diff
@@ -1,20 +1,332 @@
+        if
+            job == xi.job.DNC and
+            player:getCharVar("dancerCompletedAF") == 7
+        then
+            return true
+        end
+
+        -- Scholar has a var for AF complete
+        if
+            job == xi.job.SCH and
+            player:getCharVar("AF_SCH_COMPLETE") == 1
+        then
+            return true
+        end
+        
+    end
+
+    return false
+end
+
+function onArtifactQuesetReset(player, job)
+    -- onEventFinish is sanity checking params - so we dont have to here
+    player:delGil(10000)
+    player:delQuest(afQuestInfo[job].questcity0, afQuestInfo[job].quest0)
+    player:delQuest(afQuestInfo[job].questcity1, afQuestInfo[job].quest1)
+    player:delQuest(afQuestInfo[job].questcity2, afQuestInfo[job].quest2)
+
+    if job == xi.job.BLU then
+        player:setCharVar("[BLUAF]Current", 0)
+        player:setCharVar("[BLUAF]CraftingStage", 0)
+        player:setCharVar("[BLUAF]Remaining", 7)
+        player:setCharVar("[BLUAF]PaymentDay", 0)
+        player:setCharVar("[BLUAF]RestingDay", 0)
+    elseif job == xi.job.COR then
+        player:setCharVar("LeleroonsLetterGreen", 0)
+        player:setCharVar("LeleroonsLetterBlue", 0)
+        player:setCharVar("LeleroonsLetterRed", 0)
+    elseif job == xi.job.PUP then
+        player:setCharVar("[PUPAF]Current", 0)
+        player:setCharVar("[PUPAF]Remaining", 7)
+        player:setCharVar("[PUPAF]TradeDone", 0)
+        player:setCharVar("[PUPAF]TradeDay", 0)
+    elseif job == xi.job.DNC then
+        player:setCharVar("dancerCompletedAF", 0)
+        player:setCharVar("dancerTailorCS", 0)
+        player:setCharVar("dancerAFChoice", 0)
+    elseif job == xi.job.SCH then
+        player:setCharVar("AF_SCH_COMPLETE", 0)
+        player:setCharVar("AF_Loussaire", 0)
+        player:setCharVar("AF_SCH_BOOTS", 0)
+        player:setCharVar("AF_SCH_PANTS", 0)
+        player:setCharVar("AF_SCH_BODY", 0)
+    elseif job < xi.job.BLU then -- All Dark Spark AF jobs
+        player:delQuest(afQuestInfo[job].questcity3, afQuestInfo[job].quest3)
+    end
+end
+
 entity.onTrade = function(player, npc, trade)
 end
 
 entity.onTrigger = function(player, npc)
-    player:startEvent(10034)
+    local gil = player:getGil()
+    -- Filter/Param0 - bitmask that hides what is matched
+    --   1 hides "I have changed my mind".  2 hides War.  4 hides Mnk.  6 hides Mnk and War.
+    --   1 << xi.job = mask for the job
+
+    local filter = 8388606 -- this hides everything but "I have changed my mind".  Noteably - this also hides GEO and RUN.
+
+    for i = xi.job.WAR, xi.job.SCH do
+        if checkArtifactProgress(player, i) then
+            filter = filter - (bit.lshift(1, i))
+        end
+    end
+
+    player:startEvent(10034, filter, gil)
 end
 
 entity.onEventUpdate = function(player, csid, option)
 end
 
 entity.onEventFinish = function(player, csid, option)
+    if
+        option >= 1 and
+        option <= 20
```

Comment:
should add if gil > 10000

Reply:
Or is that handled in the options / event?

Reply:
gil is passed via event params - the client will short circuit the event if less than 10k is given as a param